adds link to our dev docs (#26)
sourcegraph/deploy: doc/development.md

## Building an AMI

> To create an AMI for a given T-shirt size, follow the [instructions in our development docs](https://docs.sourcegraph.com/admin/deploy/machine-images/aws-ami#deploy-sourcegraph) and refer to the following tables.

Reviewer: ```suggestion
To create an AMI for a given T-shirt size, please refer to the following tables and follow the instructions below.
```
The deploy docs you linked is for deploying an instance using AMIs and not for creating/building the AMI. The development docs is referring to the same page so I can see why it is confusing 😅
